src/ir/ty.rs

```diff
@@ -612,7 +612,7 @@ impl Type {
                             TypeKind::TemplateAlias(inner.unwrap(), args)
                         }
                         CXCursor_TemplateRef => {
-                            let referenced = location.referenced();
+                            let referenced = location.referenced().unwrap_or(location);
```

Comment:
Yes, we should use either `unwrap` or expect, if the cursor here is invalid, it's probably a bug. If it isn't, then we can return early an `Err` here.